azurerm/internal/services/network/resource_arm_virtual_network.go

```diff
@@ -434,8 +470,21 @@ func resourceAzureSubnetHash(v interface{}) int {
 
 	if m, ok := v.(map[string]interface{}); ok {
 		buf.WriteString(m["name"].(string))
-		buf.WriteString(m["address_prefix"].(string))
-
+		if v, ok := m["address_prefixes"]; ok {
+			switch v.(type) {
+			case []string:
+				for _, a := range v.([]string) {
+					buf.WriteString(a)
+				}
+			case []interface{}:
+				for _, a := range v.([]interface{}) {
+					buf.WriteString(a.(string))
+				}
+			}
```

Comment:
are you seeing the address_prefixes to be 2 different types that you need the switch?

Reply:
Yes! I am unsure why, but it shows up as []string and []interface{}.

Reply:
do you have a clearer idea as to when that happens? i.e when each case is activated?

Reply:
Not off of the top of my head, but I can easily find out.

Reply:
Okay, the culprit was this code (that was me) in flattenVirtualNetworkSubnets(): 

                              if prefixes := props.AddressPrefixes; prefixes != nil {
                                        var addressPrefixes []string
                                        for _, prefix := range *prefixes {
                                                addressPrefixes = append(addressPrefixes, prefix)
                                        }
                                        output["address_prefixes"] = addressPrefixes
                                }

Changing addressPrefixes to []interfaces{} got rid of the []string case .I have adjusted accordingly.